crates/starlight/src/vm/interpreter.rs

```diff
@@ -566,7 +565,7 @@ pub unsafe fn eval(mut ctx: GcPointer<Context>, frame: *mut CallFrame) -> Result
 
                 let left = lhs.to_uint32(ctx)?;
                 let right = rhs.to_uint32(ctx)?;
-                frame.push(JsValue::new((left >> (right & 0x1f)) as f64));
+                frame.push(JsValue::new(left.wrapping_shr(right) as f64));
```

Comment:
you don't have to do `as f64` JsValue::new` does work with u32 and i32, same applies for other shift operations

Reply:
Oh, I will fix it